### PR TITLE
feat(protocol-designer): add high priority analytics events for batch edit form interactions

### LINKED ISSUE #7403: PD(analytics): Add high priority analytics events for batch edit form interactions
# Overview
Add analytics events to track the following batch edit fields:

```
Select / Deselect All
Delete
Duplicate
Expand / Collapse
Flow rate Asp. (batch edit mode save step)
Tip Position Asp. (batch edit mode save step)
Well Order Asp. (batch edit mode save step)
Pre-Wet Tip Asp. (batch edit mode save step)
Mix Asp. (batch edit mode save step)
Delay Asp. (batch edit mode save step)
Touch Tip (batch edit mode save step)
Air Gap (batch edit mode save step)
Flow rate Disp. (batch edit mode save step)
Tip Position Disp. (batch edit mode save step)
Well Order Disp. (batch edit mode save step)
Delay Disp. (batch edit mode save step)
Mix Disp. (batch edit mode save step)
Touch Tip Disp. (batch edit mode save step)
Blowout Disp. (batch edit mode save step)
Air Gap Disp. (batch edit mode save step)
Exit Batch Mode Button
```

See [this sheet](https://docs.google.com/spreadsheets/d/1VbOgI4Z6je9xzpApmAt4foCa3z3gKdqmL3Yc6EGS74s/edit#gid=0) for more details

# Implementation details
We currently track analytics events via middleware (see `reduxActionToAnalyticsEvent`) that looks at an action type and sends an event to mixpanel. Add a case for `SAVE_STEP_FORMS_MULTI` in `reduxActionToAnalyticsEvent` and grab the advanced settings from `action.payload.editedFields`. 

# Acceptance criteria
I think Mixpanel events are only captured in production when `process.env.OT_PD_MIXPANEL_ID` is set, so I'm not sure how to verify the events get captured correctly. @IanLondon do we have a dev environment we can verify on? Maybe in staging?

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview
This PR adds high priority analytics events for batch edit form interactions. 

Closes #7403 

# Changelog

- Add high priority analytics events for batch edit form interactions

# Review requests
Code review, can't really check this works properly until we deploy :/

# Risk assessment
Low

### CODE REVIEW

File: protocol-designer/src/analytics/middleware.js [hunk 1]
```diff
@@ -56,6 +57,52 @@ export const reduxActionToAnalyticsEvent = (
...
+      name: 'expandMultipleSteps',
+      properties: {},
+    }
+  }
+  if (action.type === 'COLLAPSE_MULTIPLE_STEPS') {
+    return {
+      name: 'collapseMultipleSteps',
+      properties: {},
+    }
+  }
```
Comment (shlokamin) [inline protocol-designer/src/analytics/middleware.js:105]: Until we have more info on what metadata we want to pass along to mixpanel for these events, I'm leaving `properties` empty.

File: protocol-designer/src/components/StepSelectionBanner/index.js [hunk 2]
```diff
@@ -23,7 +23,10 @@ export const StepSelectionBanner = (): React.Node => {
   const dispatch = useDispatch()
 
   const { confirm, showConfirmation, cancel } = useConditionalConfirm(
-    () => dispatch(stepActions.deselectAllSteps()),
+    () =>
+      dispatch(
+        stepActions.deselectAllSteps('EXIT_BATCH_EDIT_MODE_BUTTON_PRESS')
```
Comment (shlokamin) [inline protocol-designer/src/components/StepSelectionBanner/index.js:28]: feeding in a `meta` argument (see the new signature of `deselectAllSteps `) is a bit of a hack but I think it gets the job done. def open to other ways of doing this tho

File: protocol-designer/src/ui/steps/actions/actions.js [hunk 3]
```diff
@@ -174,4 +192,23 @@ export const deselectAllSteps = (): ThunkAction<SelectStepAction> => (
...
+
+    dispatch(analyticsEvent(exitBatchEditModeEvent))
+  } else {
+    const deselectAllStepsEvent: AnalyticsEvent = {
+      name: 'deselectAllSteps',
+      properties: {},
+    }
+
+    dispatch(analyticsEvent(deselectAllStepsEvent))
+  }
```
Comment (shlokamin) [inline protocol-designer/src/ui/steps/actions/actions.js:213]: don't love this but don't know another way around bcuz there are no redux actions that are a 1:1 mapping to what we want to track

**Review by IanLondon — APPROVED** [review]
nice, I think this is correct use of `analyticsEvent` since none of the redux actions really represent what we want

**Review by Kadee80 — APPROVED** [review]
⚓

**Pull request merged**